edge-functions/ab-testing-statsig/pages/[bucket].tsx

```diff
@@ -0,0 +1,221 @@
+  }
+}
+
+export const getStaticPaths: GetStaticPaths<{ bucket: string }> = async () => {
+  // Groups that we want to statically generate
+  const groups: string[] = await api.getGroups()
+
+  return {
+    paths: groups.map((group) => ({
+      params: {
+        bucket: group,
+      },
+    })),
+    fallback: 'blocking',
+  }
+}
+
+function BucketPage({ bucket }: Props) {
+  const { reload } = useRouter()
+
+  function resetBucket() {
+    Cookie.remove('uid')
+    Statsig.logEvent('reset-bucket')
+    reload()
+  }
+
+  useEffect(() => {
+    // Log exposure to statsig
+    api.logExposure(Cookie.get('uid')!, bucket, 'statsig_example')
+  }, [bucket])
+
+  return (
+    <Page className="flex flex-col gap-12">
+      <Head>
+        <title>AB testing with Statsig - Vercel Example</title>
+        <meta
+          name="description"
+          content="Vercel example how to use ab-testing-statsig"
+        />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+
+      <section className="flex flex-col gap-6">
+        <Text variant="h1">AB testing with Statsig</Text>
+        <Text>
+          In this demo we use Statsig&apos;s REST Api at the edge to pull
+          experiment variants and show the resulting allocation. As long as you
+          have a bucket assigned you will always see the same result, otherwise
+          you will be assigned a bucket to mantain the odds specified in the
+          experiment.
+        </Text>
+        <Text>
+          Buckets are statically generated at build time in a{' '}
+          <Code>/[bucket]</Code> page so its fast to rewrite to them.
+        </Text>
+        <Snippet>{`import { NextResponse } from 'next/server'
+
+import statsig from "../api"
+
+// Store a cookie for the user
+const UID_COOKIE = 'uid'
+
+export async function middleware(req) {
+  // Clone the URL
+  const url = req.nextUrl.clone()
+
+  // Just run for the / path
+  if (req.nextUrl.pathname !== '/') {
+    return NextResponse.next()
+  }
+
+  // Get users UID from the cookie
+  let userID = req.cookies[UID_COOKIE]
+
+  // Set a userID if not present
+  if (!userID) {
+    userID = crypto.randomUUID()
+  }
+
+  // Fetch experiment
+  const bucket = await statsig.getExperiment({ userID }, 'statsig_example')
+
+  // Change the pathname to point to the correct bucket
+  url.pathname = \`/\${bucket}\`
+
+  // Create a response
+  const response = NextResponse.rewrite(url)
+
+  // Set cookie if not present
+  if (!req.cookies[UID_COOKIE]) {
+    response.cookie(UID_COOKIE, userID)
+  }
+
+  // Return the response
+  return response
+}
+`}</Snippet>
+        <Text>
+          Once the page is fully functional, we have to log the exposure for the
+          experiment, this will let Statsig know that the bucket was correctly
+          assigned and the user has been exposed to the experiment. If we
+          don&apos;t log the exposure, all users will get the same bucket and
+          the traffic won&apos;t be split.
```

Comment:
users will all get the same experiment if the traffic doesn't get reported to Statsig? why is that?